develop-docs/api-server/application-domains/database-migrations/index.mdx

```diff
@@ -145,144 +145,202 @@ This is complicated due to our deploy process. When we deploy, we run migrations
 
 To avoid this, follow these steps:
 
-- Mark the column as nullable if it isn't, and create a migration. (ex. `BoundedIntegerField(null=True)`)
-- Deploy.
-- Remove the column from the model, but in the migration make sure we only mark the state as removed.
-- Deploy.
-- Finally, create a migration that deletes the column.
+- (Optional, but ideal) Make a PR to remove all uses of the column in the codebase in a separate PR. This mostly helps with code cleanliness. This should be merged ahead of the migration prs, but we don't need to worry about whether it is deployed first.
+- Make another PR that:
+    - Checks if the column is either not nullable, or doesn't have a db_default set. If either of these is true, then make it nullable via `null=True`.
+    - If the column is a foreign key, remove the database level foreign key constraint it by setting `db_constraint=False`.
+    - Remove the column and in the generated migration use `SafeRemoveField(..., deletion_action=DeletionAction.MOVE_TO_PENDING)` to replace `RemoveField(...)`. This only marks the state for the column as removed.
+    - Combine these migrations together to save making multiple deploys
+- Deploy. It's important that all previous prs are in production before we remove the actual column from the table.
```

Comment:
Should we call this out even more explicitly, maybe a `<Note>` type thing?

Reply:
I'm not sure if it's necessary, people should just be following these steps in order, and we've had the docs like this for a while